### PR TITLE
update hugo to 0.123.8

### PR DESCRIPTION
**Pull request by Arhell**
## Reviewers

- [ ] Ambient
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [x] Developer Infrastructure
- [ ] Localization/Translation

resolve https://github.com/istio/istio.io/issues/14567

### CODE REVIEW

**Review by ericvn — CHANGES_REQUESTED** [review]
Hold out until after istio.io branch cut as noted in the issue.

**Comment by ericvn** [comment]
https://github.com/istio/tools/blob/release-1.21/docker/build-tools/Dockerfile#L56

**Comment by ericvn** [comment]
And we'll probably want to move to 0.123.1 (tooling for master branch just went to 0.123.0).

**Comment by ericvn** [comment]
netlify error Is very close to the one I fixed earlier:
```
4:54:07 PM: Start building sites …
4:54:07 PM: hugo v0.123.2-929b91fe75cb0d041f22b4707700dfc117115ad4+extended linux/amd64 BuildDate=2024-02-22T15:27:15Z VendorInfo=gohugoio
4:54:20 PM: Total in 12603 ms
4:54:20 PM: Error: error building site: render: failed to render pages: render of "page" failed: "/opt/build/repo/layouts/blog/single.html:33:11": execute of template failed: template: blog/single.html:33:11: executing "main" at <.Content>: error calling Content: "/opt/build/repo/content/zh/blog/2023/istio-graduates-within-cncf/index.md:1:1": execute of template failed: template: _default/_markup/render-link.html:1:14: executing "_default/_markup/render-link.html" at <urls.Parse>: error calling Parse: parse "(https://sessionize.com/istiocon-2023)": first path segment in URL cannot contain colon
4:54:20 PM: make: *** [Makefile.core.mk:155: netlify] Error 1
```

File: netlify.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
     publish = "public"
 
 [build.environment]
-    HUGO_VERSION = "0.121.2"
+    HUGO_VERSION = "0.123.7"
```
Comment (ericvn) [inline netlify.toml:5]:
```suggestion
    HUGO_VERSION = "0.123.8"
```

Comment (Arhell) [reply netlify.toml]: Done

**Review by ericvn — APPROVED** [review]

**Pull request merged**